File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h> // including math to use round function

int main(void)
{
    // variables declaration
    float dollars;
    int cents;
    int quarters = 0;
    int dimes = 0;
    int nickels = 0;
    int pennies = 0;

    // set the user to enter the right variable
    do
    {

        dollars = get_float("Change owed: ");
        cents = round(dollars * 100);

    }
    while (cents < 1);

    //loops to check number of quarters, dimes, nickels, and pennies.

    while (cents >= 25)
    {
        cents = cents - 25;
        quarters++; // make it as counter to add them in the end.
    }

    while (cents >= 10 && cents < 25)
    {
        cents = cents - 10;
        dimes++; // make it as counter to add them in the end.
    }

    while (cents >= 5 && cents < 10)
    {
        cents = cents - 5;
        nickels++; // make it as counter to add them in the end.
    }

    while (cents >= 1 && cents < 5)
    {
        cents = cents - 1;
        pennies++; // make it as counter to add them in the end.
    }

    int coins =  quarters + dimes + nickels + pennies; // add all the counters

    printf("%i\n", coins); // show the result

}

File: cash.py
# to use get float
from cs50 import get_float

# set the user to enter the right variable
while True:
    dollars = get_float("Change owed: ")
    cents = round(dollars * 100)
    if cents > 0:
        break

# make counters to add them in the end.
quarters = 0
dimes = 0
nickels = 0
pennies = 0
while cents >= 25:
    cents -= 25
    quarters += 1

# loops to check number of quarters, dimes, nickels, and pennies.
while cents >= 10 and cents < 25:
    cents -= 10
    dimes += 1

while cents >= 5 and cents < 10:
    cents -= 5
    nickels += 1

while cents >= 1 and cents < 5:
    cents -= 1
    pennies += 1

# add all the counters and show the result
print(quarters + dimes + nickels + pennies)



